store/cachekv/mergeiterator.go

```diff
@@ -132,7 +142,9 @@ func (iter *cacheMergeIterator) Value() []byte {
 
 	// If cache is invalid, get the parent value.
 	if !iter.cache.Valid() {
-		return iter.parent.Value()
+		value := iter.parent.Value()
+		iter.eventManager.EmitResourceAccessReadEvent("iterator", iter.storeKey, iter.parent.Key(), value)
```

Comment:
it might be cleaner to just remove the emit logic in memiterator and always emit here in the mergeiterator

Reply:
make senses, was there a reason why we only added it to the memiterator at the first place?

Reply:
I added it there, it was a misunderstanding. We should just always emit